bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/handler/EtherRainHandler.java

```diff
@@ -0,0 +1,211 @@
+            updateJob.cancel(false);
+        }
+
+        this.updateJob = null;
+    }
+
+    @SuppressWarnings("null")
+    private boolean updateBridge() {
+        if (!connected || device == null) {
+            connected = connectBridge();
+            if (!connected || device == null) {
+                connected = false;
+                device = null;
+                logger.debug("Could not connect to Etherrain device.");
+                return false;
+            }
+        }
+
+        EtherRainStatusResponse response;
+
+        try {
+            response = device.commandStatus();
+        } catch (EtherRainException | IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Could not create a connection to the EtherRain");
+            logger.debug("Could not open API connection to the EtherRain device. Exception received: {}",
+                    e.getMessage());
+            device = null;
+            return false;
+        }
+
+        updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_STATUS,
+                new StringType(response.getOperatingStatus().name()));
+
+        updateState(EtherRainBindingConstants.CHANNEL_ID_COMMAND_STATUS,
+                new StringType(response.getLastCommandStatus().name()));
+
+        switch (response.getLastCommandResult()) {
+            case OK:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_RESULT, new StringType("OK"));
+                break;
+            case RN:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_RESULT, new StringType("RAIN INTERRUPTED"));
+                break;
+            case SH:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_RESULT,
+                        new StringType("INTERRUPPTED SHORT"));
+                break;
+            case NC:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_RESULT, new StringType("DID NOT COMPLETE"));
+                break;
+        }
+
+        updateState(EtherRainBindingConstants.CHANNEL_ID_RELAY_INDEX, new DecimalType(response.getLastActiveValue()));
+
+        OnOffType rs = OnOffType.from(response.isRainSensor());
+
+        updateState(EtherRainBindingConstants.CHANNEL_ID_SENSOR_RAIN, rs);
+
+        logger.debug("Completed Etherrain Update");
+
+        return true;
+    }
+
+    private synchronized boolean execute() {
+        EtherRainCommunication device = this.device;
+
+        if (device != null) {
+            device.commandIrrigate(config.programDelay, config.zoneOnTime1, config.zoneOnTime2, config.zoneOnTime3,
+                    config.zoneOnTime4, config.zoneOnTime5, config.zoneOnTime6, config.zoneOnTime7, config.zoneOnTime8);
+            updateBridge();
+        }
+
+        return true;
+    }
+
+    private boolean clear() {
+        this.device.commandClear();
```

Comment:
Just to make the null checker happy.
```suggestion
        EtherRainCommunication device = this.device;
        if(device != null){
            device.commandClear();
        }
```